extensions/interactions/base_test.py

```diff
@@ -265,124 +265,74 @@ def test_html_field_types_to_rule_specs_mapping_are_valid(self):
+                for html_type in html_types_in_description:
+                    if 'Html' in html_type and html_type not in html_types:
+                        html_types.append(html_type)
+
                 for html_type in html_types:
-                    if html_type['type'] in description:
-                        if (
-                                html_type['type'] not in
-                                generated_html_field_types_dict):
-                            generated_html_field_types_dict[
-                                html_type['type']] = {}
-                        generated_html_field_types_dict[
-                            html_type['type']]['interactionId'] = (
-                                interaction_id)
-                        generated_html_field_types_dict[html_type['type']][
-                            'format'] = html_type['format']
-                        if (
-                                'ruleTypes' not in
-                                generated_html_field_types_dict[
-                                    html_type['type']]):
-                            generated_html_field_types_dict[html_type['type']][
-                                'ruleTypes'] = {}
-                        generated_html_field_types_dict[html_type['type']][
-                            'ruleTypes'][rule_type] = {}
-                        generated_html_field_types_dict[html_type['type']][
-                            'ruleTypes'][rule_type]['htmlInputVariables'] = (
-                                ['x'])
-                        if '{{y|%s}}' % html_type['type'] in description:
-                            generated_html_field_types_dict[html_type['type']][
-                                'ruleTypes'][rule_type][
-                                    'htmlInputVariables'].append('y')
+                    if html_type in description:
+                        if html_type not in generated_html_field_types_dict:
+                            generated_html_field_types_dict[html_type] = {}
+                        html_type_dict = (
+                            generated_html_field_types_dict[html_type])
+                        html_type_dict['interactionId'] = interaction_id
+                        if 'ruleTypes' not in html_type_dict:
+                            html_type_dict['ruleTypes'] = {}
+                        rule_types_dict = html_type_dict['ruleTypes']
+                        rule_types_dict[rule_type] = {}
+                        is_input_variables_key_created = (
+                            'htmlInputVariables' in rule_types_dict[rule_type])
+                        if not is_input_variables_key_created:
+                            rule_types_dict[rule_type]['htmlInputVariables'] = (
+                                [])
+                        # Extract the input variables of the rule specs from the
+                        # description.
+                        input_variables_with_html_type = (
```

Comment:
I don't understand why you are doing this in the for loop, since it doesn't seem to depend on the html_type in question. Couldn't you just calculate it once, outside? Also, how do you know that it's "html type"?

I would expect to see one re.findall() at the outset that collects the input variable and their types, and then filters these pairs as needed.

Reply:
done